__tests__/app.test.js

```diff
@@ -155,12 +154,65 @@ describe('/api/articles/:article_id/comments', () => {
                 expect(response.body.msg).toBe('Bad request');
             })
     });
-    test('GET:404 responds with error message when no comments', () => {
+    test('GET:200 responds with empty array when no comments', () => {
         return request(app)
             .get('/api/articles/8/comments')
-            .expect(404)
+            .expect(200)
             .then((response) => {
-                expect(response.body.msg).toBe('Not found');
+                expect(response.body.comments).toEqual([]);
+            })
+    });
+    test('POST:201 responds with the posted comment', () => {
+        return request(app)
+            .post('/api/articles/8/comments')
+            .send({
+                username: 'rogersop',
+                body: 'yeah right'
+            })
+            .expect(201)
+            .then((response) => {
+                expect(response.body.comment).toEqual(expect.objectContaining({
+                    comment_id: expect.any(Number),
+                    created_at: expect.any(String),
+                    author: 'rogersop',
+                    body: 'yeah right',
+                    article_id: 8
+                }))
+            })
+    });
+    test('POST:400 send error when required data not included in body', () => {
+        return request(app)
+            .post("/api/articles/8/comments")
+            .send({
+                body: 'no way'
             })
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe("Bad request");
+            });
     });
+    test('POST:404 returns error message when username doesnt exist',() => {
+        return request(app)
+            .post('/api/articles/8/comments')
+            .send({
+                username: 'idontexist',
+                body: 'yeah right'
+            })
+            .expect(404)
+            .then((response) => {
+                expect(response.body.msg).toBe("Not found");
+            });
+    })
+    test('POST:404 returns error message when article doesnt exist',() => {
+        return request(app)
+            .post('/api/articles/88888/comments')
+            .send({
+                username: 'rogersop',
+                body: 'yeah right'
+            })
+            .expect(404)
+            .then((response) => {
+                expect(response.body.msg).toBe("Not found");
+            });
+    })
```

Comment:
This is also excellent to see, but dont forget we should also be checking if the article id is invalid i.e. /api/articles/banana/comments